lib/floki/selector.ex

```diff
@@ -103,13 +103,14 @@ defmodule Floki.Selector do
   defp id_match?(%HTMLNode{attributes: []}, _), do: false
   defp id_match?(%HTMLNode{type: :pi}, _), do: false
 
-  defp id_match?(%HTMLNode{attributes: attributes}, id) do
-    Enum.any?(attributes, fn attribute ->
-      case attribute do
-        {"id", ^id} -> true
-        _ -> false
-      end
-    end)
+  defp id_match?(%HTMLNode{attributes: attributes}, id) when is_list(attributes) do
+    id_attr_value = :proplists.get_value("id", attributes, nil)
+    id_attr_value == id
+  end
+
+  defp id_match?(%HTMLNode{attributes: attributes}, id) when is_map(attributes) do
```

Comment:
Not sure if this  would change anything, but one possible optimization is to check if `"id"` is a key inside the `attributes` map:

```suggestion
  defp id_match?(%HTMLNode{attributes: attributes}, id) when is_map_key(attributes, "id") do
```
This already verifies if the term is a map. But you would need a default clause below.

Reply:
Changed to use pattern match to get the value directly a2172ab9fa446117369e51bbf1f92f56a1f4d834